src/themer.tsx

```diff
@@ -7,40 +7,64 @@ import { Fabric } from 'office-ui-fabric-react/lib/Fabric';
 import { Customizer } from 'office-ui-fabric-react/lib/Utilities';
 import { IPartialTheme, createTheme } from 'office-ui-fabric-react/lib/Styling';
 import { SCOPED_SETTINGS, THEME_DARK, THEME_LIGHT } from './app/constants/themes';
+import { Theme, MonacoTheme, ThemeContextProvider } from './app/shared/contexts/themeContext';
+import { THEME_SELECTION } from './app/constants/browserStorage';
 
-export enum Theme {
-    light = 'light',
-    dark = 'dark',
-}
-
-interface ThemeState {
+interface ThemerState {
     theme: Theme;
-    officeTheme: IPartialTheme;
+    fabricTheme: IPartialTheme;
+    monacoTheme: MonacoTheme;
 }
+export default class Themer extends React.Component<{}, ThemerState> {
 
-export default class Themer extends React.Component<{}, ThemeState> {
-
+    // tslint:disable-next-line: cyclomatic-complexity
     public constructor(props: {}) {
         super(props);
+        let theme = localStorage.getItem(THEME_SELECTION);
+        if (!theme) {
+            theme = document.body.classList.contains('theme-dark') ? Theme.dark : Theme.light;
+        }
         this.state = {
-            officeTheme: THEME_LIGHT,
-            theme: Theme.light
+            fabricTheme: theme === Theme.dark ? THEME_DARK : THEME_LIGHT,
+            monacoTheme: theme === Theme.dark ? MonacoTheme.dark : MonacoTheme.light,
```

Comment:
why do fabricTheme and monacoTheme need to be in state? we can just compute them in ThemeContextProvider